### PR TITLE
fix: backport datagrid missing cells issue

### PR DESCRIPTION
**Pull request by bdryanovski**
Backport of this PR #4787

### CODE REVIEW

File: packages/angular/projects/clr-angular/src/data/datagrid/wrapped-column.ts [hunk 1]
```diff
@@ -27,6 +27,15 @@ export class WrappedColumn implements DynamicWrapper, AfterViewInit, OnDestroy {
...
-    this.columnView.destroy();
+    /**
+     * @TODO
+     * Check to see if Ivy is enabled and if so call destroy method.
+     * This is related to this issues #4692 #4232
+     *
+     * This check is for backward compatibility for NON-Ivy builds
+     */
+    if (this[`__ngContext__`]) {
+      this.columnView.destroy();
```
Comment (mathisscott) [inline packages/angular/projects/clr-angular/src/data/datagrid/wrapped-column.ts:38]: Would it make sense to make this a separate helper function as opposed to repeating the conditional in three places?

That would also make it easier to unit test in isolation...

```
function checkIfIvyAndThen(context, fn) {
    if(context[`__ngContext__`] {
       fn();
    }
}
```

...and...

```
ngOnDestroy() {
    checkIfIvyAndThen(this, () => {
        this.columnView.destroy();
    }
}
```

Comment (bdryanovski) [reply packages/angular/projects/clr-angular/src/data/datagrid/wrapped-column.ts:38]: Give me a few minutes .. to find a place to put it in...

File: packages/angular/projects/clr-angular/src/data/datagrid/wrapped-cell.ts [hunk 2]
```diff
@@ -25,6 +25,15 @@ export class WrappedCell implements DynamicWrapper, AfterViewInit, OnDestroy {
...
   ngOnDestroy() {
-    this.cellView.destroy();
+    /**
+     * @TODO
+     * Check to see if Ivy is enabled and if so call destroy method.
+     * This is related to this issues #4692 #4232
+     *
+     * This check is for backward compatibility for NON-Ivy builds
+     */
+    if (this[`__ngContext__`]) {
```
Comment (Shijir) [inline packages/angular/projects/clr-angular/src/data/datagrid/wrapped-cell.ts:35]: I looked in the Angular source to see whether they offer a way for users to check Ivy is enabled or not. Apparently they do have [privately export module](https://github.com/angular/angular/blob/master/packages/core/src/core_private_export.ts#L21) for that, which means it's not exactly public API. But it's still better than what we are doing here with `__ngContext__`.  So basically we could do:

```ts
import { ..., ɵivyEnabled as ivyEnabled  } from '@angular/core';

...

ngOnDestroy() {
    if(ivyEnabled) {
      this.rowView.destroy();
    }
}

Comment (mathisscott) [reply packages/angular/projects/clr-angular/src/data/datagrid/wrapped-cell.ts:35]: If that's the case, then the utility function is an even easier path...

Comment (mathisscott) [reply packages/angular/projects/clr-angular/src/data/datagrid/wrapped-cell.ts:35]:
```
import { eivyEnabled as ivyEnabled } from '@angular/core';

function checkIfIvyAndThen(fn) {
    if(ivyEnabled) {
       fn();
    }
}
```
...and...

```
ngOnDestroy() {
    checkIfIvyAndThen(() => {
        this.columnView.destroy();
    }
}
```

Comment (Shijir) [reply packages/angular/projects/clr-angular/src/data/datagrid/wrapped-cell.ts:35]: Well in both cases, we are relying on Angular's private implementation detail. But the change I am suggesting relies on the exported variable that tells us directly what we want to know. `__ngContext__`, on the other hand, is completely private property.

**Comment by bdryanovski** [comment]
quickest merge .. ;)

**Pull request merged**